src/definitions.jl

```diff
@@ -561,3 +576,58 @@ Pre-plan an optimized real-input unnormalized transform, similar to
 the same as for [`brfft`](@ref).
 """
 plan_brfft
+
+##############################################################################
+
+region(p::Plan) = p.region
+region(p::ScaledPlan) = region(p.p)
+
+# Projection style (:none, :real, or :real_inv) to handle real FFTs
+function projection_style end
+# Length of halved dimension, needed only for irfft 
+function irfft_dim end
+
+mutable struct AdjointPlan{T,P} <: Plan{T}
+    p::P
+    pinv::Plan
+    AdjointPlan{T,P}(p) where {T,P} = new(p)
+    # always have adjoint inside scaled
+    AdjointPlan{T,P}(p::P) where {T,P<:ScaledPlan{T}} = ScaledPlan{T}(AdjointPlan{T}(p.p), p.scale)
+    AdjointPlan{T,P}(p::AdjointPlan{T}) where {T,P} = new(p.p)
+end
+
+AdjointPlan{T}(p::P) where {T,P} = AdjointPlan{T,P}(p)
+AdjointPlan(p::Plan{T}) where {T} = AdjointPlan{T}(p)
+Base.adjoint(p::Plan{T}) where {T} = AdjointPlan{T}(p)
```

Comment:
This could be simplified to the default fallback
```suggestion
Base.adjoint(p::Plan{T}) where {T} = AdjointPlan{T}(p, typeof(p))
```
if `AdjointPlan` constructors are an internal implementation detail and `adjoint` the API.